3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -18,18 +19,67 @@ Complete the four functions provided in the starter `index.js` file:
 Try and avoid using global variables. Instead, use function parameters and 
 return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function fetchData(url) {
+  return fetch(url).then((response) => {
+    if (response.ok) {
+      return response.json();
+    }
+    throw new Error('Request failed');
+  });
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function fetchAndPopulatePokemons(data) {
+  const listOfPokemons = data.results;
+  const body = document.querySelector('body');
+  const buttonGetPokemon = document.createElement('button');
+  buttonGetPokemon.textContent = 'Get Pokemon!';
+  body.appendChild(buttonGetPokemon);
+  const select = document.createElement('select');
+  body.appendChild(select);
+  const showListOfPokemons = () => {
+    listOfPokemons.forEach((pokemon) => {
+      const option = document.createElement('option');
+      option.textContent = pokemon.name;
+      select.appendChild(option);
+    });
+    select.addEventListener('change', (event) =>
+      fetchImage(event.target, listOfPokemons)
+    );
+
+    buttonGetPokemon.removeEventListener('click', showListOfPokemons);
```

Comment:
On the one hand, is it good that you prevent the pokemons to be added again if the button is pressed more than once. But removing the event handler does not give visual clue to the user that the button is no longer functioning. It would be better to disable the button rather than removing the event listener. That will then grey out the button, giving the user the desired visual clue.